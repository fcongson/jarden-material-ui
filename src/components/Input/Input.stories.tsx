import React, { useState } from 'react'
import { Input } from './Input'

export default {
  title: 'Input',
  component: Input
}

export const TextInput = () => {
  const [value, setValue] = useState('Lorem ipsum')
  return <Input value={value} onChange={value => setValue(value)} />
}

export const TextInputWithLabel = () => {
  const [value, setValue] = useState()
  return <Input label={'Label'} value={value} onChange={value => setValue(value)} />
}

export const TextInputWithPlaceholder = () => {
  const [value, setValue] = useState()
  return <Input placeholder={'Placeholder'} value={value} onChange={value => setValue(value)} />
}

export const TextInputWithUncontrolledValue = () => <Input defaultValue={'Lorem ipsum'} />
