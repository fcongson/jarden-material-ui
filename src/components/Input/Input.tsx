import TextField from '@material-ui/core/TextField'
import * as React from 'react'
import { MaterialStyles } from '../../styles/material'

interface InputProps {
  name?: string
  label?: string
  value?: string
  placeholder?: string
  onChange?: (value: string) => void | undefined
  onBlur?: (value: string) => void | undefined
}

export const Input: React.FC<InputProps> = ({ name, label, value, placeholder, onChange, onBlur, ...restProps }) => (
  <>
    <MaterialStyles />
    <TextField
      name={name}
      label={label}
      value={value}
      placeholder={placeholder}
      onChange={event => {
        if (!!onChange) onChange(event.target.value)
      }}
      onBlur={event => {
        if (!!onBlur) onBlur(event.target.value)
      }}
      fullWidth
      {...restProps}
    />
  </>
)
