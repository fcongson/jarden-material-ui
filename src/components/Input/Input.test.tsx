import * as React from 'react'
import { shallow } from 'enzyme'
import { Input } from './Input'

describe('Input', () => {
  it('Renders', () => {
    const input = shallow(<Input />)
    expect(input).toMatchSnapshot()
  })
})
