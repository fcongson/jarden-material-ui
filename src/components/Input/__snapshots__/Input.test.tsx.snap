// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders 1`] = `ShallowWrapper {}`;
